src/pages/organize/[orgId]/projects/[campId]/emails/[emailId]/insights.tsx

```diff
@@ -119,26 +121,56 @@ const EmailPage: PageWithLayout = () => {
       <Box display="flex" justifyContent="flex-end" mb={1}>
         <ZUIFuture future={emailsFuture}>
           {(emails) => (
-            <TextField
-              label={messages.insights.comparison.label()}
-              onChange={(ev) =>
-                setSecondaryEmailId(parseInt(ev.target.value) || 0)
+            <Autocomplete
+              filterOptions={(options, state) =>
+                options.filter(
+                  (email) =>
+                    email.title
+                      ?.toLowerCase()
+                      .includes(state.inputValue.toLowerCase()) ||
+                    email.campaign?.title
+                      .toLowerCase()
+                      .includes(state.inputValue.toLowerCase())
+                )
               }
-              select
-              size="small"
-              value={secondaryEmailId}
-            >
-              <MenuItem value={0}>
-                {messages.insights.comparison.noneOption()}
-              </MenuItem>
-              {emails
-                .filter((email) => email.id != emailId)
-                .map((email) => (
-                  <MenuItem key={email.id} value={email.id}>
-                    {email.title}
-                  </MenuItem>
-                ))}
-            </TextField>
+              getOptionLabel={(option) => option.title || ''}
+              onChange={(_, value) => setSecondaryEmailId(value?.id ?? 0)}
+              onReset={() => setSecondaryEmailId(0)}
+              options={emails.filter(
+                // Can only compare with published emails, and not itself
+                (email) => email.id != emailId && email.published
+              )}
+              renderInput={(params) => (
+                <TextField
+                  {...params}
+                  label={messages.insights.comparison.label()}
+                  size="small"
+                  variant="outlined"
+                />
+              )}
+              renderOption={(props, option) => (
+                <ListItem {...props}>
+                  <Box
+                    sx={{
+                      alignItems: 'stretch',
+                      display: 'flex',
+                      flexDirection: 'column',
+                    }}
+                  >
+                    <Typography>{option.title}</Typography>
+                    <Typography variant="body2">
+                      {option.campaign?.title}
+                    </Typography>
+                  </Box>
+                </ListItem>
+              )}
+              sx={{
+                minWidth: 300,
+              }}
+              value={
+                emails.find((email) => email.id == secondaryEmailId) || null
```

Comment:
You're right about the documentation. I don't know if this has changed in recent versions of React but I was almost 100% sure that `null` and `undefined` was treated differently.

Anyway, none of it applies here because we're not dealing (directly) with an `input` or `textarea` or whatever here. We're dealing with a MUI `Autocomplete` which follows different rules. I have tested to verify that `undefined` and `null` are treated differently and that `null` is fine (generates no warnings) while `undefined` is not (generates warning below).

![image](https://github.com/user-attachments/assets/d1f5c765-48f6-4674-8106-5f52dd5f7e0c)